File: Python/Problems/Problems/problem3876.py
import heapq
import math
import sys

from utils import print1DArray, print2DArray
from typing import List, Set, Optional
from collections import Counter

from .problem import Problem


class Problem3876(Problem):
    def test(self) -> bool:
        nums1 = [1, 4, 7]

        expected = True

        result = self.uniformArray(nums1)

        return result == expected

    def uniformArray(self, nums1: list[int]) -> bool:
        mn = nums1[0]
        odd = False
        for n in nums1:
            mn = min(mn, n)
            if n % 2 == 1:
                odd = True
        if mn % 2 == 1:
            return True
        return not  odd

File: C++/Problems/Problems/problem_3876.cpp
#include "problem_3876.h"

bool problem_3876::test() {
    std::vector nums1 { 1, 4, 7 };

    bool expected = true;

    auto result = uniformArray(nums1);

    return result == expected;
}

bool problem_3876::uniformArray(std::vector<int> &nums1) {
    int mn = nums1[0];
    bool odd = false;
    for (auto n : nums1)
    {
        mn = std::min(mn, n);
        if (n % 2 == 1)
            odd = true;
    }
    if (mn % 2 == 1)
        return true;
    return !odd;
}
